homeassistant/components/xiaomi_tv/media_player.py

```diff
@@ -88,16 +121,19 @@ def turn_off(self):
         would be unable to turn the TV back on, unless it's done manually.
         """
         if self._state is not STATE_OFF:
-            self._tv.sleep()
-
-            self._state = STATE_OFF
+            if self._tv.assume_state:
+                self._tv.sleep()
```

Comment:
Sleeping the TV includes a call to `time.sleep`.

https://github.com/simse/pymitv/blob/f0a347308d6a73fbad4ca27e82f480cedf60d6d5/pymitv/control.py#L35-L37

Sync sleep isn't allowed in our thread pool. It risks exhausting the pool. If sync sleep is necessary, it has to be done in a separate thread.

If the library would support asyncio it could use `asyncio.sleep` which is free.

Reply:
This one is already in use from xiaomi tv component. As I said I will try to rewrite pymitv module to use asyncio. But for now I think this can't change to something else.

Reply:
If we're changing this line we can't keep it since it's bad.

Reply:
I understand the reason of what you say, but till I rewrite the module with asyncio, I think that to let it this way is much better than remove it completely. (This will make it a breaking change)

Reply:
An alternative is to wait with the change around assumed state until the library behaves according to home assistant standards.